### PR TITLE
Split panel

### PR DESCRIPTION
**Pull request by bjarnef**
## Description

Split panel to split an area into a start and end area.
Currently named `<uui-split-panel>` but considered `<uui-split-view>` as well, but not sure if it will collide with the split view concept (compare language/segments) in Umbraco.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

- [ ] If my change requires a change to the documentation, I have updated the documentation in this pull request.
- [x] I have read the **[CONTRIBUTING](<(https://github.com/umbraco/Umbraco.UI/blob/dev/docs/CONTRIBUTING.md)>)** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/uui-resize-observer/lib/uui-resize-observer.element.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  private _resizeObserver!: ResizeObserver;
+  private _observedElements: HTMLElement[] = [];
+
+  /** Disables the observer. */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._resizeObserver = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+      this.dispatchEvent(new UUIResizeEvent(UUIResizeEvent.CHANGE));
```
Comment (bjarnef) [inline packages/uui-resize-observer/lib/uui-resize-observer.element.ts:29]: @iOvergaard would `CustomEvent` be better instead of a `UUIResizeEvent`?

Comment (iOvergaard) [reply packages/uui-resize-observer/lib/uui-resize-observer.element.ts:29]: We tend to use UUI-events inside the library. They ultimately extend Event. I think it's perfectly fine what you are doing here.

**Comment by bjarnef** [comment]
@nielslyngsoe I imagine something like this: https://shoelace.style/components/split-panel

image

The current backoffice allows to drag size of tree area, the new backoffice doesn't (yet).

Besides that it could eventually be used in splitview or other parts on the UI.

I also like the feature to have an initial position, horizontal/vertical, disabled, min and max size of panels.
Maybe also the nested split panels.

Currently the split view feature in Umbraco backoffice is horizontal, but no all property editor works well on a small width, e.g. Image Cropper. In that case maybe splitted vertical would be better?

**Comment by iOvergaard** [comment]
Hi @bjarnef 
We have started building a split panel component in the new backoffice repository. We'd like to keep it there for now and then eventually move it upstream to the UI library for others to enjoy.

You can check it out here: https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/1174

Resize observer could be interesting to have as a component, but not sure how well we can use it. We just implemented a resize observer on uui-tab-group so if we could generalize that a little bit, that would be interesting to explore. Do you have any thoughts in that regard?

I am going to close this draft PR and would like to see the efforts continued on the other repository, so please if you have anything to offer on that component, feel free to have a look!

**Pull request closed without merging**